src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -577,6 +610,29 @@ public AzureVMAgent createProvisionedAgent(
                                         "AzureVMCloud: createProvisionedAgent: VM available: {0}",
                                         resource);
 
+                                Map<String, Object> properties = AzureVMAgentTemplate.getTemplateProperties(template);
```

Comment:
Why not just set UAMI related parameters in the VM deployment template like MSI does, [link](https://github.com/jenkinsci/azure-vm-agents-plugin/blob/13e38daaaa4bad397e61ca37a3b5d1df28d318e6/src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java#L360-L363). No need to maintain a new json file here.

Reply:
I assume the reason is the API version currently in use by the ARM template doesn't support user assigned managed identity
https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/2016-03-30/virtualmachines

I'll see if just updating the api version works, but might be some effort involved...

Reply:
I see. There are even lots of preview version in the json files :sweat:.

Reply:
just running a deployment now, let's see how it goes, I haven't updated the API version, previously I was sending `null` and not `{}` and I was getting the expected validation errors..